File: leecode-practice/day1/day1.py
def getSumIndexArr(nums, sum):
  result = []
  for i, num in enumerate(nums):
    if i not in result: 
      complement = sum - num
      for currIndex, value in enumerate(nums):
        if value == complement:
          result.append(i)
          result.append(currIndex)
  print(result)

getSumIndexArr([1, 3, 9, 2, 8, 7, 4], 10)

File: leecode-practice/day1/day1.js
function getSumIndexArr(nums, sum){
  const result = []
  nums.forEach((num, index) => {
    if(!result.includes(index)) {
      const complement = sum - num
      const compIndex = nums.indexOf(complement)
      if(compIndex !== -1) {
        result.push(index)
        result.push(compIndex)
      }
    }
  })
  console.log(result);
} 
  
getSumIndexArr([1, 3, 9, 2, 8, 7, 4], 10)